src/interfaces/interfaces.ts

```diff
@@ -94,6 +94,7 @@ namespace interfaces {
         serviceIdentifier: interfaces.ServiceIdentifier<T>;
         key?: string | number | symbol;
         value?: any;
+        tags?: Tag[];
```

Comment:
This would be better to not be optional

Reply:
and plan and and _createTags

Reply:
we should prevent _createTags from adding metadata with no key as well

Reply:
> we should prevent _createTags from adding metadata with no key as well

Technically typescript is ensuring us that, right? Do we need to double check on `_createTags`?

Reply:
yes that's correct